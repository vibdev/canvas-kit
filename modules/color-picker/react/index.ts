import ColorInput from './lib/ColorInput';
import ColorPreview from './lib/ColorPreview';

export default ColorInput;
export {ColorInput, ColorPreview};
export * from './lib/ColorInput';
export * from './lib/ColorPreview';
