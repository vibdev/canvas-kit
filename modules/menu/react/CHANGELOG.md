# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.0.21](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-menu/compare/@workday/canvas-kit-react-menu@0.0.20...@workday/canvas-kit-react-menu@0.0.21) (2019-06-13)

**Note:** Version bump only for package @workday/canvas-kit-react-menu





## [0.0.20](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-menu/compare/@workday/canvas-kit-react-menu@0.0.19...@workday/canvas-kit-react-menu@0.0.20) (2019-06-13)

**Note:** Version bump only for package @workday/canvas-kit-react-menu





## [0.0.19](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-menu/compare/@workday/canvas-kit-react-menu@0.0.18...@workday/canvas-kit-react-menu@0.0.19) (2019-06-10)

**Note:** Version bump only for package @workday/canvas-kit-react-menu





## [0.0.18](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-menu/compare/@workday/canvas-kit-react-menu@0.0.17...@workday/canvas-kit-react-menu@0.0.18) (2019-06-05)

**Note:** Version bump only for package @workday/canvas-kit-react-menu





## [0.0.17](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-menu/compare/@workday/canvas-kit-react-menu@0.0.16...@workday/canvas-kit-react-menu@0.0.17) (2019-06-05)

**Note:** Version bump only for package @workday/canvas-kit-react-menu





## [0.0.16](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-menu/compare/@workday/canvas-kit-react-menu@0.0.15...@workday/canvas-kit-react-menu@0.0.16) (2019-06-03)

**Note:** Version bump only for package @workday/canvas-kit-react-menu





## [0.0.15](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-menu/compare/@workday/canvas-kit-react-menu@0.0.14...@workday/canvas-kit-react-menu@0.0.15) (2019-06-03)

**Note:** Version bump only for package @workday/canvas-kit-react-menu





## [0.0.14](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-menu/compare/@workday/canvas-kit-react-menu@0.0.13...@workday/canvas-kit-react-menu@0.0.14) (2019-05-31)

**Note:** Version bump only for package @workday/canvas-kit-react-menu





## [0.0.13](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-menu/compare/@workday/canvas-kit-react-menu@0.0.12...@workday/canvas-kit-react-menu@0.0.13) (2019-05-30)


### Bug Fixes

* **Card:** background color should be white ([31955bd](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-menu/commits/31955bd))





## [0.0.12](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-menu/compare/@workday/canvas-kit-react-menu@0.0.11...@workday/canvas-kit-react-menu@0.0.12) (2019-05-24)

**Note:** Version bump only for package @workday/canvas-kit-react-menu





## [0.0.11](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-menu/compare/@workday/canvas-kit-react-menu@0.0.10...@workday/canvas-kit-react-menu@0.0.11) (2019-05-22)

**Note:** Version bump only for package @workday/canvas-kit-react-menu





## [0.0.10](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-menu/compare/@workday/canvas-kit-react-menu@0.0.9...@workday/canvas-kit-react-menu@0.0.10) (2019-05-21)

**Note:** Version bump only for package @workday/canvas-kit-react-menu





## [0.0.9](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-menu/compare/@workday/canvas-kit-react-menu@0.0.8...@workday/canvas-kit-react-menu@0.0.9) (2019-05-15)

**Note:** Version bump only for package @workday/canvas-kit-react-menu





## [0.0.8](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-menu/compare/@workday/canvas-kit-react-menu@0.0.7...@workday/canvas-kit-react-menu@0.0.8) (2019-05-14)

**Note:** Version bump only for package @workday/canvas-kit-react-menu





## [0.0.7](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-menu/compare/@workday/canvas-kit-react-menu@0.0.6...@workday/canvas-kit-react-menu@0.0.7) (2019-05-10)

**Note:** Version bump only for package @workday/canvas-kit-react-menu





## [0.0.6](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-menu/compare/@workday/canvas-kit-react-menu@0.0.5...@workday/canvas-kit-react-menu@0.0.6) (2019-05-09)

**Note:** Version bump only for package @workday/canvas-kit-react-menu





## [0.0.5](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-menu/compare/@workday/canvas-kit-react-menu@0.0.4...@workday/canvas-kit-react-menu@0.0.5) (2019-05-09)


### Bug Fixes

* **common:** move storybook utils to outside folder ([1f661b6](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-menu/commits/1f661b6))





## [0.0.4](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-menu/compare/@workday/canvas-kit-react-menu@0.0.3...@workday/canvas-kit-react-menu@0.0.4) (2019-05-09)

**Note:** Version bump only for package @workday/canvas-kit-react-menu





## [0.0.3](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-menu/compare/@workday/canvas-kit-react-menu@0.0.2...@workday/canvas-kit-react-menu@0.0.3) (2019-05-07)

**Note:** Version bump only for package @workday/canvas-kit-react-menu





## [0.0.2](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-menu/compare/@workday/canvas-kit-react-menu@0.0.1...@workday/canvas-kit-react-menu@0.0.2) (2019-05-06)

**Note:** Version bump only for package @workday/canvas-kit-react-menu





## 0.0.1 (2019-04-29)

**Note:** Version bump only for package @workday/canvas-kit-react-menu
