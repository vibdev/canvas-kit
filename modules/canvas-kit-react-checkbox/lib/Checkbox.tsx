import * as React from 'react';
import styled from 'react-emotion';
import {focusRing} from '@workday/canvas-kit-react-common';
import {colors} from '@workday/canvas-kit-react-core';
import {SystemIcon} from '@workday/canvas-kit-react-icon';
import {checkSmallIcon} from '@workday/canvas-system-icons-web';
import {blockchainIcon} from '@workday/canvas-accent-icons-web';

export interface CheckboxProps extends React.HTMLAttributes<HTMLInputElement> {
  checked: boolean;
  disabled?: boolean;
  id?: string;
  onChange?: (e: React.SyntheticEvent) => void;
  value?: string;
  inputRef?: React.Ref<HTMLInputElement>;
}

const checkboxWidth = 18;
const checkboxHeight = 18;
const checkboxSpacing = 11;
const checkboxBorderRadius = 2;
const RippleRadius = 11;

const CheckboxContainer = styled('div')({
  position: 'relative',
  height: checkboxHeight,
  width: checkboxWidth,
  marginBottom: checkboxSpacing,
});

const CheckboxInput = styled('input')<CheckboxProps>(
  {
    position: 'absolute',
    height: checkboxHeight,
    width: checkboxWidth,
    borderRadius: checkboxBorderRadius,
    margin: 0,
    opacity: 0,
    '&:focus, &:active': {
      outline: 'none',
    },
    // I wonder if this should be replaced by a event where we look at if the Input is on focus/hover and then use these as props
    '&:focus, &:active, &focus:hover, &:active:hover': {
      '& ~ div:first-of-type': {
        ...focusRing(0, 0),
        borderColor: colors.blueberry400,
        borderWidth: '2px',
      },
      '&:checked ~ div:first-of-type': {
        ...focusRing(2, 2),
      },
    },
    '&:hover ~ div:nth-of-type(2)': {
      boxShadow: '0 0 0 ' + RippleRadius + 'px ' + colors.soap200,
    },
  },
  ({disabled, checked}) => ({
    '&:hover ~ div:first-of-type': {
      borderColor: checked ? colors.blueberry400 : colors.licorice200,
    },
  })
);

const CheckboxBackground = styled('div')<Pick<CheckboxProps, 'checked' | 'disabled'>>(
  {
    backgroundColor: colors.frenchVanilla100,
    boxSizing: 'border-box',
    position: 'absolute',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    pointerEvents: 'none',
    width: checkboxWidth,
    height: checkboxHeight,
    borderWidth: '1px',
    borderStyle: 'solid',
    borderRadius: checkboxBorderRadius,
    padding: '0px 2px',
    transition: 'border 200ms ease, background 200ms',
  },
  ({checked, disabled}) => ({
    borderColor: checked ? colors.blueberry400 : colors.licorice100,
    backgroundColor: checked ? colors.blueberry400 : disabled ? colors.soap100 : 'white',
    '&:hover': {
      borderColor: checked ? colors.blueberry400 : colors.licorice100,
      backgroundColor: checked ? colors.blueberry400 : disabled ? colors.soap100 : 'white',
    },
  })
);

// This should live somewhere else, maybe in canvas-kit-react-common
const Ripple = styled('div')({
  position: 'absolute',
  width: checkboxWidth,
  height: checkboxHeight,
  boxShadow: '0 0 0 0 ' + colors.soap200,
  transition: 'box-shadow 200ms ease-out',
  borderRadius: 999,
  zIndex: -999,
});

const CheckboxCheck = styled('div')<Pick<CheckboxProps, 'checked'>>(
  {
    position: 'absolute',
    display: 'flex',
    flexDirection: 'column',
    transition: 'transform 200ms ease, opacity 200ms ease',
    pointerEvents: 'none',
  },
  ({checked}) => ({
    opacity: checked ? 1 : 0,
    transform: checked ? 'scale(1)' : 'scale(0.5)',
  })
);

export default class ToggleSwitch extends React.Component<CheckboxProps> {
  public static defaultProps = {
    checked: false,
  };

  public render() {
    const {checked, disabled, id, inputRef, onChange, value, ...otherProps} = this.props;

    return (
      <CheckboxContainer>
        <CheckboxInput
          checked={checked}
          disabled={disabled}
          id={id}
          innerRef={inputRef}
          onChange={onChange}
          role="checkbox"
          tabIndex={0}
          type="checkbox"
          value={value}
          {...otherProps}
        />
        <CheckboxBackground checked={checked} disabled={disabled}>
          <CheckboxCheck checked={checked}>
            <SystemIcon icon={checkSmallIcon} color={colors.frenchVanilla100} />
          </CheckboxCheck>
        </CheckboxBackground>
        <Ripple />
      </CheckboxContainer>
    );
  }
}
