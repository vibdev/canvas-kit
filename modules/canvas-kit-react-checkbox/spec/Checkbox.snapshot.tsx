import * as React from 'react';
import Checkbox from '../lib/Checkbox';
import * as renderer from 'react-test-renderer';

describe('Checkbox Snapshots', () => {
  test('renders as expected', () => {
    const component = renderer.create(
      <Checkbox
        checked={false}
        onChange={() => {
          /* foo */
        }}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
