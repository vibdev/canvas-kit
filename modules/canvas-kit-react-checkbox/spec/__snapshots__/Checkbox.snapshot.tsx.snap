// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Snapshots renders as expected 1`] = `
.emotion-9 {
  margin-bottom: 11px;
}

.emotion-0 {
  border-radius: 2px;
  height: 24px;
  margin: 0;
  margin-top: -3px;
  position: absolute;
  width: 24px;
}

.emotion-0:focus,
.emotion-0:active {
  outline: none;
}

.emotion-0:focus ~ div:first-of-type,
.emotion-0:active ~ div:first-of-type,
.emotion-0focus:hover ~ div:first-of-type,
.emotion-0:active:hover ~ div:first-of-type {
  -webkit-animation: animation-s6lrth 100ms;
  animation: animation-s6lrth 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 0px #005cb9;
  border-color: #0875e1;
  border-width: 2px;
}

.emotion-0:focus:checked ~ div:first-of-type,
.emotion-0:active:checked ~ div:first-of-type,
.emotion-0focus:hover:checked ~ div:first-of-type,
.emotion-0:active:hover:checked ~ div:first-of-type {
  -webkit-animation: animation-72u8ku 100ms;
  animation: animation-72u8ku 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #005cb9;
}

.emotion-0::after {
  border-radius: 999px;
  box-shadow: 0 0 0 0 #f0f1f2;
  content: "";
  display: block;
  height: 18px;
  margin-top: 3px;
  -webkit-transition: box-shadow 150ms ease-out;
  transition: box-shadow 150ms ease-out;
  width: 18px;
  z-index: -999;
}

.emotion-0:hover::after {
  box-shadow: 0 0 0 11px #f0f1f2;
}

.emotion-0:hover ~ div:first-of-type {
  border-color: #0875e1;
}

.emotion-0:focus:hover ~ div:first-of-type {
  border-color: #0875e1;
}

.emotion-5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #ffffff;
  border-radius: 2px;
  border-style: solid;
  border-width: 1px;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 18px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px 2px;
  pointer-events: none;
  position: absolute;
  -webkit-transition: border 200ms ease,background 200ms;
  transition: border 200ms ease,background 200ms;
  width: 18px;
  cursor: pointer;
  border-color: #0875e1;
  background-color: #0875e1;
}

.emotion-5:hover {
  background-color: #0875e1;
  border-color: #0875e1;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  pointer-events: none;
  position: absolute;
  -webkit-transition: -webkit-transform 200ms ease,opacity 200ms ease;
  -webkit-transition: transform 200ms ease,opacity 200ms ease;
  transition: transform 200ms ease,opacity 200ms ease;
  opacity: 1;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.emotion-2 {
  display: inline-block;
}

.emotion-2 svg {
  display: block;
}

.emotion-2 .wd-icon-fill {
  fill: #ffffff;
}

.emotion-2:hover .wd-icon-fill {
  fill: #333d47;
}

.emotion-2 .wd-icon-accent {
  fill: #ffffff;
}

.emotion-2:hover .wd-icon-accent {
  fill: #333d47;
}

.emotion-2 .wd-icon-background {
  fill: transparent;
}

.emotion-2:hover .wd-icon-background {
  fill: transparent;
}

.emotion-7 {
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  display: block;
  line-height: 20px;
  margin-left: 29px;
}

<div
  className="emotion-9 emotion-10"
>
  <input
    checked={true}
    className="emotion-0 emotion-1"
    disabled={false}
    role="checkbox"
    tabIndex={0}
    type="checkbox"
  />
  <div
    checked={true}
    className="emotion-5 emotion-6"
    disabled={false}
  >
    <div
      checked={true}
      className="emotion-3 emotion-4"
    >
      <span
        className="emotion-2"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-check-small wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path fill-rule=\\"nonzero\\" d=\\"M17.314 6.828l1.06 1.061a.5.5 0 0 1 0 .707l-7.778 7.778a.5.5 0 0 1-.707 0L6.354 12.84a.5.5 0 0 1 0-.707l1.06-1.06a.5.5 0 0 1 .707 0l2.122 2.12 6.364-6.364a.5.5 0 0 1 .707 0z\\" class=\\"wd-icon-fill\\"/></g></svg>",
          }
        }
      />
    </div>
  </div>
  <label
    className="emotion-7 emotion-8"
  >
    
  </label>
</div>
`;

exports[`Checkbox Snapshots renders as expected 2`] = `
.emotion-9 {
  margin-bottom: 11px;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  pointer-events: none;
  position: absolute;
  -webkit-transition: -webkit-transform 200ms ease,opacity 200ms ease;
  -webkit-transition: transform 200ms ease,opacity 200ms ease;
  transition: transform 200ms ease,opacity 200ms ease;
  opacity: 1;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.emotion-2 {
  display: inline-block;
}

.emotion-2 svg {
  display: block;
}

.emotion-2 .wd-icon-fill {
  fill: #ffffff;
}

.emotion-2:hover .wd-icon-fill {
  fill: #333d47;
}

.emotion-2 .wd-icon-accent {
  fill: #ffffff;
}

.emotion-2:hover .wd-icon-accent {
  fill: #333d47;
}

.emotion-2 .wd-icon-background {
  fill: transparent;
}

.emotion-2:hover .wd-icon-background {
  fill: transparent;
}

.emotion-7 {
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  display: block;
  line-height: 20px;
  margin-left: 29px;
}

.emotion-0 {
  border-radius: 2px;
  height: 24px;
  margin: 0;
  margin-top: -3px;
  position: absolute;
  width: 24px;
}

.emotion-0:focus,
.emotion-0:active {
  outline: none;
}

.emotion-0:focus ~ div:first-of-type,
.emotion-0:active ~ div:first-of-type,
.emotion-0focus:hover ~ div:first-of-type,
.emotion-0:active:hover ~ div:first-of-type {
  -webkit-animation: animation-s6lrth 100ms;
  animation: animation-s6lrth 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 0px #005cb9;
  border-color: #0875e1;
  border-width: 2px;
}

.emotion-0:focus:checked ~ div:first-of-type,
.emotion-0:active:checked ~ div:first-of-type,
.emotion-0focus:hover:checked ~ div:first-of-type,
.emotion-0:active:hover:checked ~ div:first-of-type {
  -webkit-animation: animation-72u8ku 100ms;
  animation: animation-72u8ku 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #005cb9;
}

.emotion-0::after {
  border-radius: 999px;
  box-shadow: 0 0 0 0 #f0f1f2;
  content: "";
  display: block;
  height: 18px;
  margin-top: 3px;
  -webkit-transition: box-shadow 150ms ease-out;
  transition: box-shadow 150ms ease-out;
  width: 18px;
  z-index: -999;
}

.emotion-0:hover::after {
  box-shadow: 0 0 0 11px #f0f1f2;
}

.emotion-0:hover ~ div:first-of-type {
  border-color: #0875e1;
}

.emotion-0:focus:hover ~ div:first-of-type {
  border-color: #7b858f;
}

.emotion-5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #ffffff;
  border-radius: 2px;
  border-style: solid;
  border-width: 1px;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 18px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px 2px;
  pointer-events: none;
  position: absolute;
  -webkit-transition: border 200ms ease,background 200ms;
  transition: border 200ms ease,background 200ms;
  width: 18px;
  border-color: #0875e1;
  background-color: #0875e1;
}

.emotion-5:hover {
  background-color: #0875e1;
  border-color: #0875e1;
}

<div
  className="emotion-9 emotion-10"
>
  <input
    checked={true}
    className="emotion-0 emotion-1"
    disabled={true}
    role="checkbox"
    tabIndex={0}
    type="checkbox"
  />
  <div
    checked={true}
    className="emotion-5 emotion-6"
    disabled={true}
  >
    <div
      checked={true}
      className="emotion-3 emotion-4"
    >
      <span
        className="emotion-2"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-check-small wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path fill-rule=\\"nonzero\\" d=\\"M17.314 6.828l1.06 1.061a.5.5 0 0 1 0 .707l-7.778 7.778a.5.5 0 0 1-.707 0L6.354 12.84a.5.5 0 0 1 0-.707l1.06-1.06a.5.5 0 0 1 .707 0l2.122 2.12 6.364-6.364a.5.5 0 0 1 .707 0z\\" class=\\"wd-icon-fill\\"/></g></svg>",
          }
        }
      />
    </div>
  </div>
  <label
    className="emotion-7 emotion-8"
  >
    
  </label>
</div>
`;

exports[`Checkbox Snapshots renders as expected 3`] = `
.emotion-9 {
  margin-bottom: 11px;
}

.emotion-2 {
  display: inline-block;
}

.emotion-2 svg {
  display: block;
}

.emotion-2 .wd-icon-fill {
  fill: #ffffff;
}

.emotion-2:hover .wd-icon-fill {
  fill: #333d47;
}

.emotion-2 .wd-icon-accent {
  fill: #ffffff;
}

.emotion-2:hover .wd-icon-accent {
  fill: #333d47;
}

.emotion-2 .wd-icon-background {
  fill: transparent;
}

.emotion-2:hover .wd-icon-background {
  fill: transparent;
}

.emotion-7 {
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  display: block;
  line-height: 20px;
  margin-left: 29px;
}

.emotion-0 {
  border-radius: 2px;
  height: 24px;
  margin: 0;
  margin-top: -3px;
  position: absolute;
  width: 24px;
}

.emotion-0:focus,
.emotion-0:active {
  outline: none;
}

.emotion-0:focus ~ div:first-of-type,
.emotion-0:active ~ div:first-of-type,
.emotion-0focus:hover ~ div:first-of-type,
.emotion-0:active:hover ~ div:first-of-type {
  -webkit-animation: animation-s6lrth 100ms;
  animation: animation-s6lrth 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 0px #005cb9;
  border-color: #0875e1;
  border-width: 2px;
}

.emotion-0:focus:checked ~ div:first-of-type,
.emotion-0:active:checked ~ div:first-of-type,
.emotion-0focus:hover:checked ~ div:first-of-type,
.emotion-0:active:hover:checked ~ div:first-of-type {
  -webkit-animation: animation-72u8ku 100ms;
  animation: animation-72u8ku 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #005cb9;
}

.emotion-0::after {
  border-radius: 999px;
  box-shadow: 0 0 0 0 #f0f1f2;
  content: "";
  display: block;
  height: 18px;
  margin-top: 3px;
  -webkit-transition: box-shadow 150ms ease-out;
  transition: box-shadow 150ms ease-out;
  width: 18px;
  z-index: -999;
}

.emotion-0:hover::after {
  box-shadow: 0 0 0 11px #f0f1f2;
}

.emotion-0:hover ~ div:first-of-type {
  border-color: #7b858f;
}

.emotion-0:focus:hover ~ div:first-of-type {
  border-color: #7b858f;
}

.emotion-5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #ffffff;
  border-radius: 2px;
  border-style: solid;
  border-width: 1px;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 18px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px 2px;
  pointer-events: none;
  position: absolute;
  -webkit-transition: border 200ms ease,background 200ms;
  transition: border 200ms ease,background 200ms;
  width: 18px;
  border-color: #a1aab3;
  background-color: #f6f7f8;
}

.emotion-5:hover {
  background-color: #f6f7f8;
  border-color: #a1aab3;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  pointer-events: none;
  position: absolute;
  -webkit-transition: -webkit-transform 200ms ease,opacity 200ms ease;
  -webkit-transition: transform 200ms ease,opacity 200ms ease;
  transition: transform 200ms ease,opacity 200ms ease;
  opacity: 0;
  -webkit-transform: scale(0.5);
  -ms-transform: scale(0.5);
  transform: scale(0.5);
}

<div
  className="emotion-9 emotion-10"
>
  <input
    checked={false}
    className="emotion-0 emotion-1"
    disabled={true}
    role="checkbox"
    tabIndex={0}
    type="checkbox"
  />
  <div
    checked={false}
    className="emotion-5 emotion-6"
    disabled={true}
  >
    <div
      checked={false}
      className="emotion-3 emotion-4"
    >
      <span
        className="emotion-2"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-check-small wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path fill-rule=\\"nonzero\\" d=\\"M17.314 6.828l1.06 1.061a.5.5 0 0 1 0 .707l-7.778 7.778a.5.5 0 0 1-.707 0L6.354 12.84a.5.5 0 0 1 0-.707l1.06-1.06a.5.5 0 0 1 .707 0l2.122 2.12 6.364-6.364a.5.5 0 0 1 .707 0z\\" class=\\"wd-icon-fill\\"/></g></svg>",
          }
        }
      />
    </div>
  </div>
  <label
    className="emotion-7 emotion-8"
  >
    
  </label>
</div>
`;

exports[`Checkbox Snapshots renders as expected 4`] = `
.emotion-9 {
  margin-bottom: 11px;
}

.emotion-2 {
  display: inline-block;
}

.emotion-2 svg {
  display: block;
}

.emotion-2 .wd-icon-fill {
  fill: #ffffff;
}

.emotion-2:hover .wd-icon-fill {
  fill: #333d47;
}

.emotion-2 .wd-icon-accent {
  fill: #ffffff;
}

.emotion-2:hover .wd-icon-accent {
  fill: #333d47;
}

.emotion-2 .wd-icon-background {
  fill: transparent;
}

.emotion-2:hover .wd-icon-background {
  fill: transparent;
}

.emotion-7 {
  color: #494949;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  display: block;
  line-height: 20px;
  margin-left: 29px;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  pointer-events: none;
  position: absolute;
  -webkit-transition: -webkit-transform 200ms ease,opacity 200ms ease;
  -webkit-transition: transform 200ms ease,opacity 200ms ease;
  transition: transform 200ms ease,opacity 200ms ease;
  opacity: 0;
  -webkit-transform: scale(0.5);
  -ms-transform: scale(0.5);
  transform: scale(0.5);
}

.emotion-0 {
  border-radius: 2px;
  height: 24px;
  margin: 0;
  margin-top: -3px;
  position: absolute;
  width: 24px;
}

.emotion-0:focus,
.emotion-0:active {
  outline: none;
}

.emotion-0:focus ~ div:first-of-type,
.emotion-0:active ~ div:first-of-type,
.emotion-0focus:hover ~ div:first-of-type,
.emotion-0:active:hover ~ div:first-of-type {
  -webkit-animation: animation-s6lrth 100ms;
  animation: animation-s6lrth 100ms;
  box-shadow: 0 0 0 0px #ffffff,0 0 0 0px #005cb9;
  border-color: #0875e1;
  border-width: 2px;
}

.emotion-0:focus:checked ~ div:first-of-type,
.emotion-0:active:checked ~ div:first-of-type,
.emotion-0focus:hover:checked ~ div:first-of-type,
.emotion-0:active:hover:checked ~ div:first-of-type {
  -webkit-animation: animation-72u8ku 100ms;
  animation: animation-72u8ku 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #005cb9;
}

.emotion-0::after {
  border-radius: 999px;
  box-shadow: 0 0 0 0 #f0f1f2;
  content: "";
  display: block;
  height: 18px;
  margin-top: 3px;
  -webkit-transition: box-shadow 150ms ease-out;
  transition: box-shadow 150ms ease-out;
  width: 18px;
  z-index: -999;
}

.emotion-0:hover::after {
  box-shadow: 0 0 0 11px #f0f1f2;
}

.emotion-0:hover ~ div:first-of-type {
  border-color: #7b858f;
}

.emotion-0:focus:hover ~ div:first-of-type {
  border-color: #0875e1;
}

.emotion-5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #ffffff;
  border-radius: 2px;
  border-style: solid;
  border-width: 1px;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 18px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px 2px;
  pointer-events: none;
  position: absolute;
  -webkit-transition: border 200ms ease,background 200ms;
  transition: border 200ms ease,background 200ms;
  width: 18px;
  cursor: pointer;
  border-color: #a1aab3;
  background-color: white;
}

.emotion-5:hover {
  background-color: white;
  border-color: #a1aab3;
}

<div
  className="emotion-9 emotion-10"
>
  <input
    checked={false}
    className="emotion-0 emotion-1"
    disabled={false}
    role="checkbox"
    tabIndex={0}
    type="checkbox"
  />
  <div
    checked={false}
    className="emotion-5 emotion-6"
    disabled={false}
  >
    <div
      checked={false}
      className="emotion-3 emotion-4"
    >
      <span
        className="emotion-2"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-check-small wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path fill-rule=\\"nonzero\\" d=\\"M17.314 6.828l1.06 1.061a.5.5 0 0 1 0 .707l-7.778 7.778a.5.5 0 0 1-.707 0L6.354 12.84a.5.5 0 0 1 0-.707l1.06-1.06a.5.5 0 0 1 .707 0l2.122 2.12 6.364-6.364a.5.5 0 0 1 .707 0z\\" class=\\"wd-icon-fill\\"/></g></svg>",
          }
        }
      />
    </div>
  </div>
  <label
    className="emotion-7 emotion-8"
  >
    
  </label>
</div>
`;
